metrics/lib/common.bash

```diff
@@ -267,7 +267,9 @@ function kill_processes_before_start() {
 	result=$(check_active_process "$HYPERVISOR_PATH")
 	if [[ $result -ne 0 ]]; then
 		warning "Found unexpected ${HYPERVISOR_PATH} processes present"
-		pgrep -a -f "$HYPERVISOR_PATH"
+		# Sometimes we race and the process has gone by the time we list
+		# it - so make a pgrep fail non-fatal
+		pgrep -a -f "$HYPERVISOR_PATH" || true
```

Comment:
If this races due to the process(es) exiting, won't the `killall` below also fail (so doesn't that also need `|| true`?)

Reply:
you're probably right - let me check the man for killall, but I'll probably add it anyway. thx

Reply:
Yep, even with the `sudo` wrapper I think you are right, the killall could fail (although I have not see that in testing (mostly due to pgrep failing first)). Fixed, pushed.